### PR TITLE
Updated changelog with API changes in #286

### LINKED ISSUE #747: removed methods from the exported `Node` type
Document the removal of public methods in #286 in changelog

### PR DESCRIPTION
**Pull request by sergio-mena**
Closes #747

Documents all the public methods removed from `Node` and modified in `Switch`, as part of #286.

---

#### PR checklist

- [ ] Tests written/updated
- [x] Changelog entry added in `.changelog` (we use [unclog](https://github.com/informalsystems/unclog) to manage our changelog)
- [ ] Updated relevant documentation (`docs/` or `spec/`) and code comments

### CODE REVIEW

**Review by lasarojc — APPROVED** [review]

**Comment by sergio-mena** [comment]
Changing the base branch to `v0.38.x`. Since this change was backported, its changelog should refer to the first version where it appears

File: .changelog/unreleased/breaking-changes/286-p2p.md [hunk 1]
```diff
@@ -0,0 +1,2 @@
+- `[p2p]` Modified method `AddReactor` in `Switch` type: it is no longer returning the reactor
```
Comment (cason) [inline .changelog/unreleased/breaking-changes/286-p2p.md:1]:
```suggestion
- `[p2p]` Modified method `AddReactor` in `Switch` type: it is no longer returning the added `Reactor`
```

**Review by cason — APPROVED** [review]

**Review by thanethomson — APPROVED** [review]
You'd still need to run `unclog build > CHANGELOG.md` to propagate these changes to the changelog, otherwise they won't be immediately visible to alpha release users.

**Comment by sergio-mena** [comment]
> You'd still need to run unclog build > CHANGELOG.md to propagate these changes to the changelog, otherwise they won't be immediately visible to alpha release users.

Just pushed the result. There are some further changes, but I'm assuming those are some previous backports where we didn't run unclog.

**Pull request merged**